Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -65,6 +67,21 @@ sub _releases {
 		# map to role's name for readability
 		$_->{role_ids} = join(',', map { Slim::Schema::Contributor->roleToType($_) } split(',', $_->{role_ids} || ''));
 
+		my $genreMatch = undef;
+		if ( !( $menuMode && $menuMode ne 'artists' && $menuRoles ) && $prefs->get('showComposerReleasesbyAlbum')==2 ) {
+			my $requestRef = [ 'genres', 0, MAX_ALBUMS, "album_id:".$_->{id} ];
+			my $request = Slim::Control::Request->new( undef, $requestRef );
+			$request->execute();
+			$log->error($request->getStatusText()) if $request->isStatusError();
+			my $genres= $request->getResult('genres_loop');
+			foreach my $genre (@$genres) {
+				$genreMatch ||= grep {uc($genre->{genre}) eq $_} split(/,/, uc($prefs->get('showComposerReleasesbyAlbumGenres')));
+				last if $genreMatch;
+			}
+		}
```

Comment:
I'll see whether I can find a more elegant solution to this... using a request certainly does the job. But there must be something simpler...

Reply:
We could amend the abumsquery to use a subquery to return an array of genres, conditioned by a new tag.